migrations/Version20240310150000.php

```diff
@@ -0,0 +1,671 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\Migrations;
+
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+use Doctrine\DBAL\Platforms\PostgreSQLPlatform;
+use Doctrine\DBAL\Platforms\SqlitePlatform;
+use Doctrine\DBAL\Schema\Schema;
+use Wallabag\Doctrine\WallabagMigration;
+
+final class Version20240310150000 extends WallabagMigration
+{
+    public function getDescription(): string
+    {
+        return 'Fix schema';
+    }
+
+    public function up(Schema $schema): void
+    {
+        switch (true) {
+            case $this->platform instanceof MySQLPlatform:
+                $this->write('Align database schema with Doctrine metadata.');
+
+                $this->addSql('ALTER TABLE ' . $this->getTable('oauth2_clients') . ' CHANGE name name LONGTEXT NOT NULL;');
+
+                $this->addSql('ALTER TABLE ' . $this->getTable('user') . ' CHANGE salt salt VARCHAR(255) DEFAULT NULL;');
+                $this->addSql('ALTER TABLE ' . $this->getTable('user') . ' CHANGE password password VARCHAR(255) NOT NULL;');
+                $this->addSql('ALTER TABLE ' . $this->getTable('user') . ' CHANGE googleAuthenticatorSecret googleAuthenticatorSecret VARCHAR(255) DEFAULT NULL;');
+
+                $this->addSql('ALTER TABLE ' . $this->getTable('entry') . ' CHANGE published_by published_by LONGTEXT DEFAULT NULL COMMENT \'(DC2Type:array)\';');
+                $this->addSql('ALTER TABLE ' . $this->getTable('entry') . ' CHANGE headers headers LONGTEXT DEFAULT NULL COMMENT \'(DC2Type:array)\';');
+                $this->addSql('ALTER TABLE ' . $this->getTable('entry') . ' CHANGE hashed_url hashed_url VARCHAR(40) DEFAULT NULL;');
+                $this->addSql('ALTER TABLE ' . $this->getTable('entry') . ' CHANGE hashed_given_url hashed_given_url VARCHAR(40) DEFAULT NULL;');
+                $this->addSql('ALTER TABLE ' . $this->getTable('entry') . ' CHANGE is_not_parsed is_not_parsed TINYINT(1) DEFAULT 0 NOT NULL;');
+
+                $this->addSql('ALTER TABLE ' . $this->getTable('tag') . ' CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_bin;');
+                $this->addSql('ALTER TABLE ' . $this->getTable('tag') . ' CHANGE label label LONGTEXT NOT NULL;');
+                $this->addSql('ALTER TABLE ' . $this->getTable('tag') . ' CHANGE slug slug VARCHAR(128) NOT NULL;');
```

Comment:
@j0k3r @nicosomb @Kdecherf what to do for columns that became NOT NULL? should we assume that there is always a value already and it's OK as is? should I add a statement that will fill with and empty string?
it's important because as migrations are not running in a transaction because of a bug on PHP8, it can fail in the middle...

Reply:
@j0k3r @Kdecherf I hope end user databases and wallabag.it ones don't have NULL value in columns becoming NOT NULL in the schema.
but what if it's the case? this is my only concern about this migration (even before the time it will take to run)

Reply:
How can we prevent it?
There should have a slug for all of them. But just in case, some guardrails might be better.

Reply:
I will abort the migration if there are malformed rows.

A summary of new not null constraint:

mysql:
- wallabag_tag.label
- wallabag_tag.slug
- wallabag_entry.is_not_parsed
- wallabag_annotation.text

postgresql:
- wallabag_entry.is_not_parsed
- wallabag_site_credential.id
- wallabag_ignore_origin_user_rule.id
- wallabag_ignore_origin_instance_rule.id

sqlite:
- wallabag_entry.is_not_parsed

Reply:
I don't see any case (_even bogus_) that would lead to a null value for `id` columns.

For `is_not_parsed`, I'm seeing in the original migration that the default value should be `false` thus I don't think we could trigger an issue here.

Finally for `tag`.`label`, `tag`.`slug` and `annotation`.`text` I would be for a precondition check before running the migration: if there is any null value in these columns then the migration is aborted and the admin must handle the case by hand. What do you think @yguedidi, @j0k3r?

Reply:
> I would be for a precondition check before running the migration: if there is any null value in these columns then the migration is aborted and the admin must handle the case by hand.

It's what I meant by "adding safeguards" in https://github.com/wallabag/wallabag/pull/7220#issuecomment-2033115357, and it's already pushed 😎

Reply:
I agree for `id`s and `is_not_parsed` there shouldn't be any issue, but I'd go safe and still add the safeguard, as we are adding a NOT NULL constraint, just in case